Info module

Hariofspades opened the pull request:
* feat: Info module
* feat: team list and event overview
* chore: refactored speakers to be in common module
* chore: reusable models

Commit:
feature: speaker details

Commit:
chore: unit test for speakers module

Commit:
feat: team info

chore: New result handler, refactoring common views/controllers/epoxy models to base

Commit:
feat: event list setup

Commit:
chore: refactoring info module to have shared viewmodel

Commit:
feat: team details and event overview

chore: Refactoring to common module

RivuChk commented on .idea/gradle.xml:
```diff
@@ -10,7 +10,9 @@
             <option value="$PROJECT_DIR$" />
```
This file shouldn't be changed, can you check .gitignore? Or remove this file manually from changelist?

Hariofspades replied:
crazy. I duno why these files got pushed anyway. Will check it

RivuChk replied:
Updated .gitignore, this shouldn't be an issue next time onwards

RivuChk commented on .idea/misc.xml:
```diff
@@ -1,11 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
This file shouldn't be changed, can you check .gitignore? Or remove this file manually from changelist?

RivuChk commented on .idea/navEditor.xml:
```diff
@@ -1,14 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project version="4">
```
This file shouldn't be changed, can you check .gitignore? Or remove this file manually from changelist?

RivuChk commented on .idea/vcs.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
This file shouldn't be changed, can you check .gitignore? Or remove this file manually from changelist?

RivuChk commented on speakers/src/main/java/in/droidcon/speakers/ui/list/SpeakerListFragment.kt:
```diff
@@ -1,20 +1,21 @@
-package `in`.droidcon.speakers.presentation.ui
```
Why does info module implementation PR has speakers module changes? Can we have these speakers module changes as separate PR?

RivuChk commented on speakers/src/main/res/layout/fragment_speaker_details.xml:
```diff
@@ -0,0 +1,31 @@
+<androidx.constraintlayout.widget.ConstraintLayout
```
Same here, can we have these Speaker Details related changes in the Speaker Detail PR itself? (I believe it's not merged yet, so you can update that PR), so that we can separate concerns in the PR

Hariofspades replied:
the problem is, There are a lot of refactoring for speakers module. I have moved epoxy and DTOs to base for this module, so that team, sponsors and speakers + sponsor detail, speaker-detail and team can reuse stuff. It's better to keep it incremental. As more refactoring might come if we work on sponsors.

RivuChk commented on speakers/src/test/java/in/droidcon/speakers/domain/GetAllSpeakersTest.kt:
```diff
@@ -0,0 +1,44 @@
+package `in`.droidcon.speakers.domain
```
Same here, please create separate PR for Speaker module / speaker detail module related changes.

Hariofspades replied:
Same scenario - the problem is, There are a lot of refactoring for speakers module. I have moved epoxy and DTOs to base for this module, so that team, sponsors and speakers + sponsor detail, speaker-detail and team can reuse stuff. It's better to keep it incremental. As more refactoring might come if we work on sponsors.

RivuChk reviewed (changes requested):
- Can you attach of screenshot?
- Are all the base module, speaker module changes required for info module implementations?

RivuChk committed:
Updated .gitignore
Can't remove previously added unwanted .idea/ files without re-writing history, they should be carefully avoided from commit in future
